src/rgw/rgw_multi.cc

```diff
@@ -245,3 +245,20 @@ int abort_multipart_upload(RGWRados *store, CephContext *cct, RGWObjectCtx *obj_
   return ret == -ENOENT?-ERR_NO_SUCH_UPLOAD:ret;
 }
 
+int list_bucket_multiparts(RGWRados *store, RGWBucketInfo& bucket_info,
+				string& prefix, string& marker, string& delim,
+				RGWAccessListFilter *filter, int& max_uploads,
+				vector<rgw_bucket_dir_entry> *objs,
+				map<string, bool> *common_prefixes, bool *is_truncated)
+{
+  RGWRados::Bucket target(store, bucket_info);
+  RGWRados::Bucket::List list_op(&target);
+
+  list_op.params.prefix = prefix;
+  list_op.params.delim = delim;
+  list_op.params.marker = marker;
+  list_op.params.ns = RGW_OBJ_NS_MULTIPART;
+  list_op.params.filter = filter;
```

Comment:
i'm guessing that all callers of `list_bucket_multiparts()` will want to use a `MultipartMetaFilter`. can we just put one on the stack here so callers don't have to pass one in?
```c++
  MultipartMetaFilter filter;
  list_op.params.filter = &filter;
```